log4j-core-test/src/test/java/org/apache/logging/log4j/core/LoggerMessageFactoryCustomizationTest.java

```diff
@@ -24,34 +24,36 @@
 import org.apache.logging.log4j.message.MessageFactory;
 import org.apache.logging.log4j.message.ParameterizedMessageFactory;
 import org.junit.jupiter.api.Test;
-import org.junitpioneer.jupiter.ClearSystemProperty;
 import org.junitpioneer.jupiter.SetSystemProperty;
 
+@SetSystemProperty(
```

Comment:
By making this test-scoped, aren't you changing the `arguments_should_be_honored()` test? That is,

* Before: there are no property-provided configurations, but only arguments passed to `Logger::new`
* New: there are property-provided configurations, and also arguments passed to `Logger::new`

In the new case, you cannot know which configuration source has kicked in.

Reply:
Fixed in https://github.com/apache/logging-log4j2/pull/3263/commits/3c3e29b4a2b730885dca4885ff0b15facebb10bb.

Reply:
@ppkarwasz, can you help me to understand how did you implement the test case where there are no property-provided configurations?

Reply:
Are you suggesting that there is a missing `getClass()` and `isEqualTo` in the assertions? You are right, nice catch! :100:

Reply:
I would like to see tests for these two cases:

1. Passing `MF`s as arguments to `Logger::new` (all `MF`-related properties are cleared before the test)
2. Passing `MF`s as properties

Can you point me the lines where you do these checks, please?

Reply:
Fixed in 8ee4237af8e86b5404f40fea2e8db2952096b28b

Reply:
> 1. Passing `MF`s as arguments to `Logger::new` (all `MF`-related properties are cleared before the test)

This PR moved the initialization of the default `MessageFactory` and `FlowMessageFactory` to the **static** initialization of the `LoggerContext` class. Clearing the `MF`-related properties before the test doesn't change anything.

The PR also reverts the (IMHO overly-specific) Javadoc for the `Logger` constructors introduced in https://github.com/apache/logging-log4j2/commit/3efd59e49323038c05d8db9d912e850cd0b23134, with a simple "the `MessageFactory` and `FlowMessageFactory` parameters should not be `null`. `null` is allowed for BC".

Reply:
Default `MF`s are configured using properties. When I pass `MF`s to `Logger::new`, I want the passed arguments to override the defaults. Where do you test this?

Reply:
https://github.com/apache/logging-log4j2/blob/551652a1d7cdef1becd915fc15e291483c15b317/log4j-core-test/src/test/java/org/apache/logging/log4j/core/LoggerMessageFactoryCustomizationTest.java#L39-L47

The configuration properties point to the `Alternative*` implementations.

Reply:
Gotcha. That indeed does the trick. Thanks.